File: BACKEND/RestAPI_net/Admin/Models/Donor.cs
using System;
using System.Collections.Generic;

namespace Admin.Models;

public partial class Donor
{
    public int Did { get; set; }

    public int? Uid { get; set; }

    public DateOnly Dob { get; set; }

    public string? Gender { get; set; }

    public int Bcid { get; set; }

    public string? MedicalHistory { get; set; }

    public virtual BloodComponent Bc { get; set; } = null!;

    public virtual ICollection<DonorDonation> DonorDonations { get; set; } = new List<DonorDonation>();

    public virtual User? UidNavigation { get; set; }
}

File: BACKEND/RestAPI/src/main/java/com/example/demo/entities/Donor.java
package com.example.demo.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity
@Getter
@Setter
public class Donor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer did;

    private String dob;
    private String gender;
    private String medical_history;

    // ✅ Correct Foreign Key Mapping
    @ManyToOne
    @JoinColumn(name = "bcid", nullable = false)
    private BloodComponent bloodComponent;

    @OneToOne
    @JoinColumn(name = "uid")
    private Users user;
}
